bin/ch/CMakeLists.txt

```diff
@@ -69,14 +69,20 @@ if(STATIC_LIBRARY)
       ${ICULIB}
       "-framework CoreFoundation"
       "-framework Security"
+      "-stdlib=libstdc++" # expected 'deprecated' warning. needed for std::string
       )
   endif() # Linux ?
-else() # // !from shared library
+else() # // shared library below
   set(lib_target
     PRIVATE Chakra.Pal
     PRIVATE Chakra.Common.Codex.Singular
     PRIVATE Chakra.Runtime.PlatformAgnostic.Singular
     )
+  if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    set(lib_target "${lib_target}"
+      "-stdlib=libstdc++" # expected 'deprecated' warning. needed for std::string
+      )
+  endif()
```

Comment:
Why not move this out of "if staticlib else" to top level, so that you don't need to duplicate?